nft_ingester/src/api/dapi/search_assets.rs

```diff
@@ -22,14 +22,14 @@ use super::asset_preview::populate_previews;
 pub async fn search_assets<TPF: TokenPriceFetcher>(
     index_client: Arc<impl postgre_client::storage_traits::AssetPubkeyFilteredFetcher>,
     rocks_db: Arc<Storage>,
-    filter: SearchAssetsQuery,
+    mut filter: SearchAssetsQuery,
     sort_by: AssetSorting,
     limit: u64,
     page: Option<u64>,
     before: Option<String>,
     after: Option<String>,
     cursor: Option<String>,
-    options: SearchAssetsOptions,
+    options: GetByMethodsOptions,
```

Comment:
Why do we introduce new struct `GetByMethodsOptions` and use it here instead of `SearchAssetsOptions`.
First time I thought it will be used only in `getBy...` requests but if use it for `search` query as well why not we just add one more field to `SearchAssetsOptions` ?
Is there any issues with that that I don't see?

Reply:
SearchAssetsOptions does not need to contain the `show_fungible` field, while GetByMethodsOptions need to. We use json-rpc lib, which validates input parameters, so we use different structs for allowing setting different options for different methods